### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by vpetrovykh**

### CODE REVIEW

File: docs/edgeql/statements/with.rst [hunk 1]
```diff
@@ -166,36 +166,25 @@ An example of incorrect refactoring would be:
...
-
-For example, the following query will find all users who
-own the same number of issues as someone else:
-
-.. todo::
-
-    Need a better DETACHED example, with nested sub-queries and
-    possibly motivated by keeping the symbols closer to their place if
-    usage.
+A ``DETACHED`` expression allows to refer to some set as if it were
```
Comment (elprans) [inline docs/edgeql/statements/with.rst]: *allows referring*

File: docs/edgeql/statements/with.rst [hunk 2]
```diff
@@ -166,36 +166,25 @@ An example of incorrect refactoring would be:
...
+defined in the top-level ``WITH`` block. Basically, ``DETACHED``
+expressions ignore all current scopes they are nested in and only take
+into account module aliases. The net effect is that it is possible to
+refer to an otherwise related type as if it were unrelated:
```
Comment (elprans) [inline docs/edgeql/statements/with.rst]: related *set*

File: docs/edgeql/statements/with.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/edgeql/statements/with.rst]: Fixed

File: docs/edgeql/statements/with.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/edgeql/statements/with.rst]: Fixed

File: docs/edgeql/lexical.rst [hunk 3]
```diff
@@ -92,17 +92,112 @@ They can be used in most places where a short name can appear (such as
...
+diacritics like ``ò`` or ``ü`` in the dollar-quote delimiter.
+
+Here's a list of valid :token:`str_escapes`:
+
+.. _ref_eql_lexical_str_escapes:
+
++--------------------+---------------------------------------------+
+| Escape Sequence    | Meaning                                     |
++====================+=============================================+
+| ``\newline``       | Backslash and newline ignored               |
```
Comment (1st1) [inline docs/edgeql/lexical.rst]: This isn't clear. I'd say ``\[newline]``

File: docs/edgeql/lexical.rst [hunk 4]
```diff
@@ -92,17 +92,112 @@ They can be used in most places where a short name can appear (such as
...
++--------------------+---------------------------------------------+
+| ``\t``             | ASCII tabulation (``\x09``)                 |
++--------------------+---------------------------------------------+
+| ``\xhh``           | Character with hex value hh                 |
++--------------------+---------------------------------------------+
+| ``\uhhhh``         | Character with 16-bit hex value hhhh        |
++--------------------+---------------------------------------------+
+| ``\Uhhhhhhhh``     | Character with 32-bit hex value hhhhhhhh    |
++--------------------+---------------------------------------------+
+
```
Comment (1st1) [inline docs/edgeql/lexical.rst]: Add a paragraph about raw strings.  Cross-link that paragraph from edgeql/expressions

File: docs/edgeql/statements/with.rst [hunk 5]
```diff
@@ -166,36 +166,25 @@ An example of incorrect refactoring would be:
...
-
-For example, the following query will find all users who
-own the same number of issues as someone else:
-
-.. todo::
-
-    Need a better DETACHED example, with nested sub-queries and
-    possibly motivated by keeping the symbols closer to their place if
-    usage.
+A ``DETACHED`` expression allows referring to some set as if it were
```
Comment (1st1) [inline docs/edgeql/statements/with.rst]: This should go in a separate PR.

File: docs/edgeql/lexical.rst (hunk 4 above)
Comment (1st1) [reply docs/edgeql/lexical.rst]: I'd also add a few examples of valid literals.

* trailing `\`
* multiline strings
* `$$` & `$foo$`
* raw strings

File: docs/edgeql/lexical.rst (hunk 3 above)
Comment (vpetrovykh) [reply docs/edgeql/lexical.rst]: ok

File: docs/edgeql/statements/with.rst (hunk 5 above)
Comment (vpetrovykh) [reply docs/edgeql/statements/with.rst]: OK, will make a separate PR.

File: docs/edgeql/lexical.rst (hunk 4 above)
Comment (vpetrovykh) [reply docs/edgeql/lexical.rst]: Added examples.

There's already a link from the expressions section to the constants section right after the table.
